clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -203,20 +201,19 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
 
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
-        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create())) {
             byte[] bytes = new byte[data.capacity() * 2];
             int nRead;
             while ((nRead = in.read(bytes, 0, bytes.length)) != -1) {
-                out.write(bytes, 0, nRead);
+                try (ByteBufferOutputStream out = new ByteBufferOutputStream(nRead)) {
+                    out.write(bytes, 0, nRead);
+                    return out.buffer();
```

Comment:
should we call `flip` before returning it??

Reply:
Done. Thanks!